sqlglot/transforms.py

```diff
@@ -9,6 +9,27 @@
     from sqlglot.generator import Generator
 
 
+def update_from(expression: exp.Expression) -> exp.Expression:
+    """Transform UPDATE FROM to MERGE INTO"""
+    from_expression = expression.find(exp.From)
+    if isinstance(expression, exp.Update) and from_expression:
+        join = from_expression.find(exp.Join)
+        if join:
+            on = join.args.get("on")
+            table = join.find(exp.Table)
+            this = expression.this
+            then = exp.Update(expressions=expression.expressions)
+            when = exp.When(matched=True, then=then)
+            expression = exp.Merge(this=this, using=table, on=on, expressions=[when])
+        else:
+            table = from_expression.find(exp.Table)
```

Comment:
Same here, should we instead do `from_expression.this`?

Reply:
Let me check this out